packages/data-transport-layer/src/services/l2-ingestion/service.ts

```diff
@@ -145,7 +145,22 @@ export class L2IngestionService extends BaseService<L2IngestionServiceOptions> {
           await sleep(this.options.pollingInterval)
         }
       } catch (err) {
-        if (!this.running || this.options.dangerouslyCatchAllErrors) {
+        if (err.message.includes('missing transaction entry')) {
```

Comment:
Catching the error here will cause it to retry on the missed transaction?

Reply:
Oh oops. I forgot to actually recover here.

Reply:
Hmmm we actually don't need to catch this error at all. We'll retry simply because we've thrown an error. I copy/pasted the logic from the L1 sync codepath but this is kinda unnecessary.

Reply:
I need to look at this again. We might want to try to explicitly recover by calling `setHighestSyncedUnconfirmedBlock(lastGoodTransaction.index)`` within this `catch`.